File: Engine/ScriptCore/src/Physics/PhysicsMaterial.cs
#pragma warning disable 1591
using System;

namespace Stulu
{
	public struct PhysicsMaterial {
		/// <summary>
		/// Coefficient of static friction
		/// </summary>
		public float StaticFriction;
		/// <summary>
		/// Coefficient of dynamic friction
		/// </summary>
		public float DynamicFriction;
		/// <summary>
		/// Coefficient of restitution
		/// </summary>
		public float Restitution;

		public PhysicsMaterial(float sF = 0.6f, float dF = 0.6f, float re = 0.0f)
		{
			StaticFriction = sF;
			DynamicFriction = dF;
			Restitution = re;
		}
	}
}

File: Engine/Stulu/src/Stulu/Physics/PhysicsMaterial.cpp
#include "st_pch.h"
#include "PhysicsMaterial.h"
#include "PhysicsModule.h"

#define PX_PHYSX_STATIC_LIB
#include "PxConfig.h"
#include "PxPhysicsAPI.h"

namespace Stulu {
	void PhysicsMaterial::SetStaticFriction(float value) {
		if(m_material)
			m_material->setStaticFriction(value);

		StaticFriction = value;
	}
	void PhysicsMaterial::SetDynamicFriction(float value) {
		if (m_material)
			m_material->setDynamicFriction(value);

		DynamicFriction = value;
	}
	void PhysicsMaterial::SetRestitution(float value) {
		if (m_material)
			m_material->setRestitution(value);

		Restitution = value;
	}
	void PhysicsMaterial::CreateMaterial() {
		auto& physicsModule = PhysicsModule::Get();

		if (physicsModule.PhysicsEnabled())
			m_material = physicsModule.GetPhysics()->createMaterial(StaticFriction, DynamicFriction, Restitution);
		else
			m_material = nullptr;
	}
}

